apps/judicial-system/api/src/app/modules/auth/auth.service.ts

```diff
@@ -32,18 +35,50 @@ export class AuthService {
   }
 
   async findDefender(nationalId: string): Promise<User | undefined> {
-    const res = await fetch(
-      `${this.config.backendUrl}/api/cases/limitedAccess/defender?nationalId=${nationalId}`,
-      {
-        headers: { authorization: `Bearer ${this.config.secretToken}` },
-      },
-    )
+    try {
+      const res = await fetch(
+        `${this.config.backendUrl}/api/cases/limitedAccess/defender?nationalId=${nationalId}`,
+        {
+          headers: { authorization: `Bearer ${this.config.secretToken}` },
+        },
+      )
+      if (res.ok) {
+        return await res.json()
+      }
+    } catch (error) {
+      console.warn('Defender not found', error)
```

Comment:
Please check specifically for a NotFoundException and only log and pass through for that exceptions. Also, consider logging as info. An error is logged futher down the call path "Blocking login attempt from an unauthorized user". That message should probably be logged as a warning instead.

If the exception is anything else it should be rethrown as it indicates an error on the backend or a problem with the connection to the backend.

Reply:
Please do not use console logging. Instead, inject a logger and use it instead.